### ISSUE TITLE
[Quickbooks Source] Unable to configure for multiple entity quickbooks account

### ISSUE THREAD

**Issue by CAMPSMITH** [airbytehq/airbyte#44794]
### Connector Name

source-quickbooks

### Connector Version

3.0.19

### What step the error happened?

Configuring a new connector

### Relevant information

We are having issues properly configuring our quickbooks source connector.  We have multiple business entities in our quickbooks account.  When configuring the source connector, we were unable to specify which entity would be in scope for the sync for the streams in scope.
The documentation makes no mention one way or the other related to support of multiple entities within quickbooks.
Are multiple entities supported? If so, how are the entities differentiated within a stream, e.g. is there a field in the payment stream that denotes one entity from another?  Again, the documentation makes no mention one way or the other.  We require the ability to differentiate the streams by business entity.

Any help or clarification is greatly appreciated.

Martin

### Contribute

- [ ] Yes, I want to contribute

**Comment by natikgadzhi** [comment]
Ooops! What is the actual behavior that you observed? Did it sync all entities? 

We're looking at a couple of issues with QuickBooks that we're working on — if we have the capacity, would you be willing to provide your account tokens to test with us, make sure that the changes that we make actually work?

**Comment by CAMPSMITH** [comment]
Hi Natik,

We did not yet start a sync.  This is because our targets are set up by
business entities.  We are in the process of seeing if we can set up a test
target so we can assess the output of the sync.  As you can imagine with
quickbooks, our business considers this data to be very sensitive data.  We
do not share account tokens externally.  Is there a way we could coordinate
the testing?

Cheers,
Martin

**Comment by natikgadzhi** [comment]
Hypothetically, we can best effort fix it and prepare a dev version release for you that you can setup on your Airbyte OSS, or we can pin for you on Airbyte Cloud, and you can use it. If it works — great, we merge and make the new version available for everyone. 

Just a tad higher risk to break things.

**Comment by natikgadzhi** [comment]
Given it's likely going to be a new config option, we'd need to figure out a path to make this not a breaking change for existing connections, and assume default value of "none" for account_ids.

**Comment by CAMPSMITH** [comment]
Hi Natik,

We have set up a new destination separate from pre-existing ones.  We will
enable the connector, hopefully today.  Is there any information I can
provide that would help?  We are using Airbyte Cloud.  Do you need our
Airbute workspace url?

I will meet with our accounting group, hopefully today and follow up with
some details about our quickbooks set up, in case that helps.

Thank you for the quick responses,

Martin

**Comment by natikgadzhi** [comment]
Yeah, you can just post a blob on the behavior you see (does it sync everything from all accounts? just the first? Drops data?) DM me a link to your workspace on our community slack (same username there, `@natikgadzhi`).

I can take a look at the logs, and from there I can ping you once we get to this with a dev version of the connector.

**Comment by CAMPSMITH** [comment]
Hi Natik,

Unfortunately, we were unable to complete the OAuth set up for our
Quickbooks prod environment.  We were successful in the sandbox.  The prod
environment requires a set up we don't have, like an application URL,
EULA,  and such.  We were hoping to set up an integration between our
quickbooks account and our data warehouse.  It seems that our use case
might not be what was intended with this connector, given the auth flow.

We are now considering other options.

I sincerely appreciate your rapid responses.  If you feel we should still
be a fit, please let me know.

Cheers,

Martin

**Comment by octavia-squidington-iii** [comment]
At Airbyte, we seek to be clear about the project priorities and roadmap. This issue has not had any activity for 180 days, suggesting that it's not as critical as others. It's possible it has already been fixed. It is being marked as stale and will be closed in 20 days if there is no activity. To keep it open, please comment to let us know why it is important to you and if it is still reproducible on recent versions of Airbyte.

**Comment by octavia-squidington-iii** [comment]
This issue was closed because it has been inactive for 20 days since being marked as stale.
